php/exceptions/UploadException.class.inc

```diff
@@ -0,0 +1,80 @@
+<?php declare(strict_types=1);
+/**
+ * This class is a wrapper that takes an upload error code
+ * an adds the appropriate message to the exception.
+ *
+ * PHP Version 7
+ *
+ * @category Errors
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+
+/**
+ * The UploadException class
+ *
+ * @category Errors
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+class UploadException extends \LorisException
+{
+    /**
+     * Contructor that takes the upload error code to generate the message.
+     *
+     * @param int $code The error code
+     *
+     * @see http://php.net/manual/en/features.file-upload.errors.php
+     */
+    public function __construct(int $code)
+    {
+        $message = $this->_codeToMessage($code);
+        parent::__construct($message, $code);
+    }
+
+    /**
+     * Mapper function to retreive the corresponding error message.
+     *
+     * @param int $code The error code
+     *
+     * @return string
+     */
+    private function _codeToMessage(int $code) : string
+    {
+        switch ($code) {
+        case UPLOAD_ERR_INI_SIZE:
+            $message = 'The uploaded file exceeds the upload_max_filesize ' .
+                       'directive in php.ini';
+            break;
+        case UPLOAD_ERR_FORM_SIZE:
+            $message = 'The uploaded file exceeds the MAX_FILE_SIZE ' .
+                       'directive that was specified in the HTML form';
+            break;
+        case UPLOAD_ERR_PARTIAL:
+            $message = 'The uploaded file was only partially uploaded';
+            break;
+        case UPLOAD_ERR_NO_FILE:
+            $message = 'No file was uploaded';
+            break;
+        case UPLOAD_ERR_NO_TMP_DIR:
+            $message = 'Missing a temporary folder';
+            break;
+        case UPLOAD_ERR_CANT_WRITE:
+            $message = 'Failed to write file to disk';
+            break;
+        case UPLOAD_ERR_EXTENSION:
+            $message = 'File upload stopped by extension';
```

Comment:
@cmadjar says:

> not sure I understand this error. Is it because the extension of the uploaded file is not zip, tar.gz, tar or tgz? If so, maybe rephrase to: "Wrong extension of the file to upload"? Or something like that?

Reply:
Same here.

Also I think we should just go by MIME type. We can check extensions on the front-end if we want to but they don't really matter as long as the actual content is good.

Reply:
Mime type or not, we need to handle this error code. 
I think I copy&pasted that https://www.php.net/manual/en/features.file-upload.errors.php#89374

Reply:
Also, reading the description, this is not related to the file extension but rather to a php extension in php.ini

Reply:
OK so then there are no changes to make?

Reply:
If @cmadjar asked, then we should probably clarify. It could be `File upload stopped by a php extension`